robottelo/test.py

```diff
@@ -148,11 +148,16 @@ def setUp(self):  # noqa
             self._docker_browser = DockerBrowser()
             self._docker_browser.start()
             self.browser = self._docker_browser.webdriver
+            self.addCleanup(self._docker_browser.stop)
         else:
             self.browser = browser()
+            self.addCleanup(self.browser.quit)
```

Comment:
Actually whenever an exception happens the execution will stop immediately and if that happens when creating a browser, then ti will stop before even adding the cleanup method.

About having more then one failure, an example follows:

```python
import unittest


class SampleTestCase(unittest.TestCase):
    def raiseexception(self):
        raise Exception()

    def setUp(self):
        self.addCleanup(self.raiseexception)

    def test_1(self):
        pass
```


```
================================ test session starts ================================
platform linux2 -- Python 2.7.10, pytest-2.9.0, py-1.4.31, pluggy-0.3.1
rootdir: /home/elyezer/code/robottelo, inifile: 
collected 1 items 

test.py F

===================================== FAILURES ======================================
_______________________________ SampleTestCase.test_1 _______________________________

self = <test.SampleTestCase testMethod=test_1>

    def raiseexception(self):
>       raise Exception()
E       Exception

test.py:15: Exception
============================= 1 failed in 0.01 seconds ==============================
```